Capstone/Feed/HomeViewController.m

```diff
@@ -0,0 +1,112 @@
+@property (weak, nonatomic) IBOutlet UISearchBar *searchBar;
+@property (strong, nonatomic) NSMutableArray *filteredColleges;
+
+@end
+
+bool isFiltered;
+
+@implementation HomeViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.tableView.dataSource = self;
+    self.tableView.delegate = self;
+    self.searchBar.delegate = self;
+    isFiltered = false;
+    self.collegeDictionaryArrays = [[NSMutableArray alloc] init];
+    [self fetchData];
+}
+
+- (void)fetchData {
+    NSURL *url = [NSURL URLWithString:@"https://api.collegeai.com/v1/api/college-list?api_key=4c4e51cca8832178dcfb29217c&filters=&info_ids=website%2CshortDescription%2ClongDescription%2CcampusImage%2Ccity%2CstateAbbr%2Caliases%2Ccolors%2ClocationLong%2ClocationLat"];
+    NSURLRequest *request = [NSURLRequest requestWithURL:url cachePolicy:NSURLRequestReloadIgnoringLocalCacheData timeoutInterval:10.0];
+    NSURLSession *session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration] delegate:nil delegateQueue:[NSOperationQueue mainQueue]];
+    NSURLSessionDataTask *task = [session dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+           if (error != nil) {
+               NSLog(@"%@", [error localizedDescription]);
+           } else {
+               NSDictionary *dataDictionary = [NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingMutableContainers error:nil];
+               self.colleges = [dataDictionary objectForKey:@"colleges"];
+               for (NSDictionary<NSString *, NSString *> *college in self.colleges) {
+                   [self.collegeDictionaryArrays addObject:college];
+               }
+               [self.tableView reloadData];
+            }}];
+    [task resume];
+}
+
+-(void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText {
+    if (searchText.length == 0) {
+        isFiltered = false;
+    } else {
+        isFiltered = true;
+        self.filteredColleges = [[NSMutableArray alloc] init];
+        for (NSDictionary<NSString *, NSString *> *college in self.colleges) {
+            NSRange collegeName = [college[@"name"] rangeOfString:searchText options:NSCaseInsensitiveSearch];
+            if (collegeName.location != NSNotFound) {
+                [self.filteredColleges addObject:college];
+            }
+        }
+    }
+    [self.tableView reloadData];
+}
+
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    if (isFiltered) {
+        return self.filteredColleges.count;
+    }
+    return self.collegeDictionaryArrays.count;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    HomeCell *cell = [tableView dequeueReusableCellWithIdentifier:@"HomeCell"];
+    if (isFiltered) {
+        cell.homeCollegeName.text = self.filteredColleges[indexPath.row][@"name"];
```

Comment:
Do you plan to implement the College class here? definitely would help in terms of reducing redundant code!